### PR TITLE
Sudden Death

### PR DESCRIPTION
**Pull request by DrewBritt**
Addresses #275, adding Sudden Death functionality.

https://github.com/apetavern/sbox-grubs/assets/43252311/b608d412-8767-4083-9d25-9560c631ff19

After a configured round delay, Sudden Death activates, lowering the position of the terrain every round. A "round" is considered a full rotation of players, and is currently determined by PlayerTurnQueue being empty (refilled after check).

Players can view the state of Sudden Death with the indicator in the bottom right, which either displays the number of rounds until Sudden Death, or "ACTIVE".

sdrounds

sdactive

Config settings have also been added to adjust the delay of Sudden Death (gr_sd_delay), the intensity of the movements (gr_sd_aggression), as well as whether all Grubs have their health set to 1 on Sudden Death start (gr_sd_onehealth). Please let me know if these (specifically the delay) should have their defaults tweaked before merge, I admittedly haven't played a lot of Worms so I'm not sure what would feel best.

### CODE REVIEW

File: code/Gamemodes/Modes/FFA/FreeForAll.cs [hunk 1]
```diff
@@ -91,6 +93,12 @@ private async Task NextTurn()
 
 		ActivePlayer.EndTurn();
 
+		if ( PlayerTurnQueue.Count == 0 )
+		{
+			RoundsUntilSuddenDeath -= 1;
```
Comment (matekdev) [inline code/Gamemodes/Modes/FFA/FreeForAll.cs:98]: Maybe we should move this inside of 

```
		if ( SuddenDeath )
			await Terrain.LowerTerrain( GrubsConfig.SuddenDeathAggression );
```

?

File: code/Gamemodes/Modes/FFA/FreeForAll.cs [hunk 2]
```diff
@@ -91,6 +93,12 @@ private async Task NextTurn()
 
 		ActivePlayer.EndTurn();
 
+		if ( PlayerTurnQueue.Count == 0 )
```
Comment (matekdev) [inline code/Gamemodes/Modes/FFA/FreeForAll.cs:96]: .Any()

File: code/Gamemodes/Modes/FFA/FreeForAll.cs [hunk 3]
```diff
@@ -311,8 +319,13 @@ private bool CheckWinConditions()
 
 	private void RotateActivePlayer()
 	{
-		if ( ActivePlayer.IsAvailableForTurn )
-			PlayerTurnQueue.Enqueue( ActivePlayer );
+		if ( PlayerTurnQueue.Count == 0 )
```
Comment (matekdev) [inline code/Gamemodes/Modes/FFA/FreeForAll.cs:322]: .Any()

File: code/Gamemodes/Modes/FFA/FreeForAll.cs (hunk 1 above)
Comment (DrewBritt) [reply code/Gamemodes/Modes/FFA/FreeForAll.cs:98]: This makes sense to me, but PlayerTurnQueue is only in FreeForAll.cs. I think Gamemode.cs should get a bit of a refactor when we write more gamemodes, but should I move PlayerTurnQueue up + implement this change for now?

**Pull request merged**